doc/proposals/image-scaffold-feature.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Image scaffold feature
+authors:
+  - "@camilamacedo86"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-12-05
+last-updated: 2019-12-05
+status: implementable
+see-also:
+  - "/proposals/this-other-neat-thing.md"  
+replaces:
+  - "/proposals/that-less-than-great-idea.md"
+superseded-by:
+  - "/proposals/our-past-effort.md"
+---
+
+# Image scaffold feature
+
+## Release Signoff Checklist
+
+- \[ x] Enhancement is `implementable`
+- \[ x] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created in [operator-sdk/doc][operator-sdk-doc]
+
+## Summary
+
+This proposal defines a new feature which allow users get the scaffold the
+code required to have an image deployed on the cluster following the good practices.  
+
+## Motivation
+
+- In many scenarios users would like to just have a Pod with the image managed by the Kind
+- Make the process to develop operators easier and more agil. 
+- Give flexibility to the users and allow them to change the code according to their needs
+
+## Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+
+### Non-Goals
+
+The idea of this proposal is provide a good facility for the users which indeed could be improved 
+in the future, however, this proposal just covers the basic requirements. In this way is a non-goal;
+allow to provide a list of images, implement the unit tests for the code generated, extra 
+configurations as for example, to define if should or not expose a service, use env vars and etc.
+
+## Proposal
+
+Add the following flags. 
+
+- Add the flag `--image` for the command `operator-sdk add api`
+- Add the flag `--image-gen` for the command `operator-sdk add controller`
+
+Add the following behaviours.
+
+1. Creating CR/CRD with image specs
+
+When the following command be executed the new flag `--image`
+
+`operator-sdk add api --api-version=app.example.com/v1alpha1 --kind=AppService --image=ngnix:latest`
+
+Then: 
+- The Kind struct will be scaffold with the image spec as follows:
+
+```go
+type AppServiceSpec struct {
+    ...
+	// Image for the AppService Kind
+	Image string `json:"image"`
+```
+- The CR will be generated with the spec `image: <valued informed>` 
+
+2. Scaffolding the secondary resources required 
+
+When the following command be executed with the new flag `--image-gen`
+
+`operator-sdk add controller --api-version=app.example.com/v1alpha1 --kind=AppService --image-gen`
+
+Then, the controller should be scaffold with:
+
+- A Deployment and Service to deploy the image in a Pod in the Reconcile. 
+- An implementation to check an update the status of the secondary resources in the CR
+- An implementation to Watch the Deployment and Service as well.  
```

Comment:
I think we could do even more here: 

- An environment variable referencing the image by SHA could be passed on to the scaffold Deployment of the Operator
- code referencing that environment variable and passing the value into the Deployment of the image in question

Reply:
Good idea 👍